image/manifest.go

```diff
@@ -71,3 +73,52 @@ func manifestLayerInfosToBlobInfos(layers []manifest.LayerInfo) []types.BlobInfo
 	}
 	return blobs
 }
+
+// chooseDigestFromManifestList parses blob as a ManifestList,
+// and returns the digest of the image appropriate for the current environment.
+func chooseDigestFromManifestList(sys *types.SystemContext, blob []byte, mt string) (digest.Digest, error) {
+	if mt != manifest.DockerV2ListMediaType && mt != imgspecv1.MediaTypeImageIndex {
+		return "", fmt.Errorf("Internal error: Trying to select an image from a non-manifest-list manifest type %s", mt)
```

Comment:
(Is this extra check worth the duplication of the list of supported MIME types? `manifest.ListFromBlob` is going to fail anyway. OTOH, yes, this is a bit more clear that it is an internal inconsistency.)